tests/test_render.py

```diff
@@ -1866,3 +1881,94 @@ def update_output(*args):
         self.assertTrue(timestamp_2.value > timestamp_1.value)
         self.assertEqual(call_count.value, 4)
         self.percy_snapshot('button-2 click again')
+
+    def test_graphs_in_tabs_do_not_share_state(self):
+        app = dash.Dash()
+
+        app.config.suppress_callback_exceptions = True
+
+        app.layout = html.Div([
+            dcc.Tabs(
+                id="tabs",
+                children=[
+                    dcc.Tab(label="Tab 1", value="tab1", id="tab1"),
+                    dcc.Tab(label="Tab 2", value="tab2", id="tab2"),
+                ],
+                value="tab1",
+            ),
+
+            # Tab content
+            html.Div(id="tab_content"),
+        ])
+        tab1_layout = [
+            html.Div([dcc.Graph(id='graph1',
+                                figure={
+                                    'data': [{
+                                        'x': [1, 2, 3],
+                                        'y': [5, 10, 6],
+                                        'type': 'bar'
+                                        }]
+                                })]),
+
+            html.Pre(id='graph1_info'),
+        ]
+
+
+        tab2_layout = [
+            html.Div([dcc.Graph(id='graph2',
+                                figure={
+                                    'data': [{
+                                        'x': [4, 3, 2],
+                                        'y': [5, 10, 6],
+                                        'type': 'bar'
+                                        }]
+                                })]),
+
+            html.Pre(id='graph2_info'),
+        ]
+
+        @app.callback(Output(component_id='graph1_info', component_property='children'),
+                    [Input(component_id='graph1', component_property='clickData')])
+        def display_hover_data(hover_data):
+            print(hover_data)
+            show_well_info = json.dumps(hover_data)
+            return show_well_info
+
+
+        @app.callback(Output(component_id='graph2_info', component_property='children'),
+                    [Input(component_id='graph2', component_property='clickData')])
+        def display_hover_data(hover_data):
+            print(hover_data)
+            show_well_info = json.dumps(hover_data)
+            return show_well_info
+
+        @app.callback(Output("tab_content", "children"), [Input("tabs", "value")])
+        def render_content(tab):
+            if tab == "tab1":
+                return tab1_layout
+            elif tab == "tab2":
+                return tab2_layout
+            else:
+                return tab1_layout
+
+        self.startServer(app)
+
+        self.wait_for_element_by_css_selector('#graph1')
+
+        (self.driver.find_elements_by_css_selector(
+            '#graph1'
+        )[0]).click()
+
+        self.wait_for_text_to_not_equal('#graph1_info', 'null')
```

Comment:
could we explicitly match the text instead of testing that it's not null? That way, we can ensure that the text being displayed from graph1 is different from the text being displayed in graph2.

Reply:
Note that the text output on click is very specific, and I was considering that it might change at some point, so that this test would fail every time the click data looks a little different.

Reply:
(which is why I thought maybe asserting that the text isn't null - which is what's happening in that bug - would be a better solution)

Reply:
I think this text is very specific and not subject to change. If it does change, then it's like a breaking change (unless its additive).

Reply:
Well, what if some data does get added at some point? Then we'd have to fix/update this test too. Wouldn't it be better if we had tests that only fail when there's an actual regression/error?

Reply:
Then we'd fix the test. It's only like 10 characters and you'll just copy and paste it based off the test failure, it will only take 5 minutes. Similar to how we, e.g. update jest tests when jest upgrades.

In the current verison of the test, I can't ensure that the text that your comparing is actually the proper clickData hover data. And from other comments, I'm surprised that the click data would be correct (since we're not actually clicking on points) and also the current test doesn't actually test that the clickData is different between the two graphs - we don't know from the test that the even isn't actually the same _same instance_ of the graph across tabs (I mentioned this in my original comment: https://github.com/plotly/dash-renderer/pull/101#discussion_r237667873).

Reply:
You're right in that we're not testing that at the moment. However, we test that the clickData is no longer null as it was before, proving that the clickData is coming in and that the click event is being bound. Previously, the events weren't being bound since it used the same instance of the first Graph component. But I see your point that it is not 100% convincing evidence. Asserting that Graph1 clickData does not equal Graph2 clickData seems like a good addition, I'll add that!

Reply:
Please just assert the actual content of `clickData`, not just that they aren't equal to each other. I want to see that the clickData that is coming in is correct.

Reply:
Tests are failing on 2.7 because the text is slightly different - there's a `u` in front of there etc. Hmm!